packages/workflow/src/utils.ts

```diff
@@ -30,3 +30,20 @@ export const deepCopy = <T>(source: T): T => {
 	return clone;
 };
 // eslint-enable
+
+export const parseJSON = <T>(
+	jsonString: string,
+	options?: { errorMessage?: string; fallbackValue?: any },
```

Comment:
This function should always return `T`. Can you please change this to 

```suggestion
	options?: { errorMessage?: string; fallbackValue?: T },
```

Reply:
what if user wants to set some falsy value in case of error?

Reply:
`null` and `undefined` will still be allowed. for any other fallback value, IMO the type of the fallback value should be the same as that of a valid value.

Reply:
and if someone wants to use a completely different type for a fallback-value, they should then have to explicitly use `parseJSON<any>(`.

Reply:
@netroy reverted this change as it causes a lot of unexpected errors

Reply:
I think the changes are expected. Those files need to use an explicit `<any>`.
I'll send you a quick PR.

Reply:
https://github.com/n8n-io/n8n/pull/4413